src/main/java/com/powsybl/openloadflow/OpenLoadFlowProvider.java

```diff
@@ -102,8 +103,15 @@ private static LoadFlowResult.ComponentResult.Status convertStatus(AcLoadFlowRes
         }
     }
 
+    private GraphConnectivityFactory<LfBus, LfBranch> getConnectivityFactory(OpenLoadFlowParameters parametersExt) {
+        return parametersExt.isNetworkCacheEnabled() && !parametersExt.getActionableSwitchesIds().isEmpty()
+                ? new NaiveGraphConnectivityFactory<>(LfBus::getNum)
+                : connectivityFactory;
+    }
+
     private LoadFlowResult runAc(Network network, LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt, Reporter reporter) {
-        AcLoadFlowParameters acParameters = OpenLoadFlowParameters.createAcParameters(network, parameters, parametersExt, matrixFactory, connectivityFactory);
+        GraphConnectivityFactory<LfBus, LfBranch> theConnectivityFactory = getConnectivityFactory(parametersExt);
```

Comment:
What do you mean by "theConnectivityFactory" ?

Reply:
this is just a different variable name to select either the factory provider in the constructor or force the naive one which is the only one that manage reconnection

Reply:
to not confuse with the connectivityFactory class attribute